Libtool fixes for cross-compilation, many past contributors/authors.
Update based on libtool-2.4.4, useful for 2.4.3 as well but not 2.4.2.

Upstream: Not applicable (Buildroot specific)

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
[Bernd: rebased for version 1.18]
Signed-off-by: Bernd Kuhls <bernd@kuhls.net>

--- a/build-aux/ltmain.sh	2014-12-16 09:51:23.068441045 -0300
+++ b/build-aux/ltmain.sh	2014-12-16 09:57:10.509430339 -0300
@@ -2687,8 +2687,8 @@
 	# was found and let the user know that the "--tag" command
 	# line option must be used.
 	if test -z "$tagname"; then
-	  func_echo "unable to infer tagged configuration"
-	  func_fatal_error "specify a tag with '--tag'"
+	  func_echo "defaulting to \`CC'"
+	  func_echo "if this is not correct, specify a tag with \`--tag'"
 #	else
 #	  func_verbose "using $tagname tagged configuration"
 	fi
@@ -4277,8 +4277,12 @@
 	  # At present, this check doesn't affect windows .dll's that
 	  # are installed into $libdir/../bin (currently, that works fine)
 	  # but it's something to keep an eye on.
-	  test "$inst_prefix_dir" = "$destdir" && \
-	    func_fatal_error "error: cannot install '$file' to a directory not ending in $libdir"
+
+	  #
+	  # This breaks install into our staging area.  -PB
+	  #
+	  #test "$inst_prefix_dir" = "$destdir" && \
+	  #  func_fatal_error "error: cannot install '$file' to a directory not ending in $libdir"
 
 	  if test -n "$inst_prefix_dir"; then
 	    # Stick the inst_prefix_dir data into the link command.
@@ -8112,7 +8112,7 @@
 	  *)
 	    if test no = "$installed"; then
 	      func_append notinst_deplibs " $lib"
-	      need_relink=yes
+	      need_relink=no
 	    fi
 	    ;;
 	  esac
@@ -10710,6 +10710,10 @@
 	    # Replace all uninstalled libtool libraries with the installed ones
 	    newdependency_libs=
 	    for deplib in $dependency_libs; do
+	      # Replacing uninstalled with installed can easily break crosscompilation,
+	      # since the installed path is generally the wrong architecture.  -CL
+	      newdependency_libs="$newdependency_libs $deplib"
+	      continue
 	      case $deplib in
 	      *.la)
 		func_basename "$deplib"
